php/libraries/Candidate.class.inc

```diff
@@ -680,115 +675,185 @@ class Candidate
     */
     static function _generateCandID(): int
     {
-        // get a database connection
-        $factory = NDB_Factory::singleton();
-        $db      = $factory->database();
-
-        // generate a candidate id
-        $candID = random_int(CANDIDATE_MIN_CANDID, CANDIDATE_MAX_CANDID);
-
-        // make sure the cand id is not used
-        while ($db->pselectOne(
-            "SELECT count(CandID) FROM candidate WHERE CandID=:CaID",
-            array('CaID' => $candID)
-        ) > 0) {
-            // pick a new candidate id
-            $candID = random_int(CANDIDATE_MIN_CANDID, CANDIDATE_MAX_CANDID);
-        }
-
-        return $candID;
+        return intval(\Candidate::_generateID('CandID'));
```

Comment:
I don't see the benefit of doing this for CandIDs. It's a non-configurable random number between 100000-999999 which has no other features or customizations. Trying to re-use the PSCID/ExternalID logic for CandIDs makes the more convoluted and the above code that's being deleted it pretty straight forward. (Maybe it could be improved by using a do/while loop instead of a while loop, but the logic of "generate a random number, check if it exists", repeat if necessary is hidden and is much harder to follow with this change.)

Reply:
I see what you're saying. I agree the old code isn't so bad. It did seem to be almost exactly the same as the `_generateID` code and it seemed valuable to use that function's approach of querying the DB just once instead of querying the DB each time a value is generated. However that might not be as much of an issue since these are randomly generated.

Do you feel like your concern could be addressed with a code comment? It could describe how the ID is generated and this would allow us to still follow the DRY rule.